compiler/crates/relay-compiler/src/build_project/artifact_content.rs

```diff
@@ -434,22 +516,42 @@ fn generate_fragment(
         )
         .unwrap();
     }
-    writeln!(content, "*/\n").unwrap();
-    writeln!(
-        content,
-        "var node/*: {}*/ = {};\n",
-        reader_node_flow_type,
-        printer.print_fragment(schema, reader_fragment)
-    )
-    .unwrap();
-    write_source_hash(config, &mut content, &source_hash).unwrap();
+
+    let node_fragment = printer.print_fragment(schema, reader_fragment);
+
+    if let TypegenLanguage::Flow = project_config.typegen_config.language { 
+        writeln!(content, "*/\n").unwrap();
+        writeln!(
+            content,
+            "var node/*: {}*/ = {};\n",
+            reader_node_flow_type,
+            node_fragment
+        )
+        .unwrap();
+    } else {
+        writeln!(content, "\n").unwrap();
+        writeln!(
+            content,
+            "var node:{} = {};\n",
+            reader_node_flow_type,
+            node_fragment
+        )
+        .unwrap();
+    }
+
+    write_source_hash(config, project_config, &mut content, &source_hash).unwrap();
 
     writeln!(content, "module.exports = node;").unwrap();
     sign_file(&content).into_bytes()
 }
 
-fn get_content_start(config: &Config) -> String {
+fn get_content_start(config: &Config, project_config: &ProjectConfig) -> String {
     let mut content = String::new();
+
+    if let TypegenLanguage::TypeScript = project_config.typegen_config.language {
+        writeln!(content, "// @ts-nocheck").unwrap();
```

Comment:
I didn't really find documentation on this, the best I found were these release notes:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#-ts-nocheck-in-typescript-files

Does this mean the file is ignored? Why would we want that?

Reply:
Just means don't typecheck anything in the file—but is still made available to consumers to read. So exported Node, or types is assumed to be valid, rather than checked to be valid.

Reply:
Come to think of it, this doesn't need to exist "at the very very top" just needs to exist before any code. So we can move this back into the generate methods. Sorry, will amend!

Reply:
This change has been made.